scripts/container_get_lines/container_get_lines.gml

```diff
@@ -0,0 +1,130 @@
+// Breaks the text to be displayed into lines consisting of segments; used to display items correctly when middle-aligned or right-aligned
```

Comment:
I completely missed this file last time.  This is where all the magic is happening!  After this second review, I'll do a more thorough look at this, along with the other math.

Reply:
One thing, I would like to cache this data somewhere.  Once we calculate it once, it could help to not run this each time.

Reply:
Though shoot, changing strings (like type writer effect, or other string changes) will have to recompute.  Hmm...

Reply:
I've added a cache. It seems to work great right now, since the only (provided) functions that change strings re-create a container every time they do. New container, new cache.
Ah, those containers may need to be deleted, but I don't think that's related to components. I'll leave that to you.